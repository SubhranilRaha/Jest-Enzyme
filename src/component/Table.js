import React, { useState } from 'react';

function Table() {
  const [data, setData] = useState([
    { name: 'John Doe', email: 'john.doe@example.com' },
    { name: 'Jane Doe', email: 'jane.doe@example.com' },
  ]);

  return (
    <table className="custom-border">
      <thead>
        <tr>
          <th>Name</th>
          <th>Email</th>
        </tr>
      </thead>
      <tbody>
        {data.map((row, index) => (
          <tr key={index}>
            <td>{row.name}</td>
            <td>{row.email}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default Table;
