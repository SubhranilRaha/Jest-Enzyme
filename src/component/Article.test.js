import Article from "./Article";
import { shallow,mount } from "enzyme";
import Table from "./Table";

describe('about page testing',()=>{
    it.skip('click count',()=>{
        let wrapper = shallow(<Article/>)
        expect(wrapper.find('.clicks-0').length).toEqual(1)
        wrapper.find('button').simulate('click')
        expect(wrapper.find('.clicks-1').length).toEqual(2)
        wrapper.find('button').simulate('click')
        expect(wrapper.find('.clicks-2').length).toEqual(1)
        wrapper.find('button').simulate('click')
        expect(wrapper.find('.clicks-3').length).toEqual(1)
    })
    it('click 3 count',()=>{
        let wrapper = shallow(<Article/>)
        expect(wrapper.find('.clicks-0').length).toEqual(1)
        wrapper.find('button').simulate('click')
        wrapper.find('button').simulate('click')
        wrapper.find('button').simulate('click')
        expect(wrapper.find('.clicks-3').length).toEqual(1)
    })
    it('dive',()=>{
        let wrapper = shallow(<Article/>)
        expect(wrapper.find(Table).dive().find('.custom-border').length).toBe(1)
    })
    it('snapshot 1',()=>{
        let wrapper = shallow(<Table/>)
        expect(wrapper).toMatchSnapshot()
    })
})
describe('mount vs shallow',()=>{
    it('snapshot 2',()=>{
        let wrapper = mount(<Article/>)
        expect(wrapper).toMatchSnapshot() 
    })
})