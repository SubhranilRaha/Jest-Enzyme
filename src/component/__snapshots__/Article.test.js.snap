// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about page testing snapshot 1 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        John Doe
      </td>
      <td>
        john.doe@example.com
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        Jane Doe
      </td>
      <td>
        jane.doe@example.com
      </td>
    </tr>
  </tbody>
</table>
`;
